src/API/Site/Controllers/RestAPI/SyncController.php

```diff
@@ -0,0 +1,147 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\RestAPI;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseOptionsController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use Exception;
+use WP_REST_Request as Request;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class SyncController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\RestAPI
+ *
+ * @since x.x.x
+ */
+class SyncController extends BaseOptionsController {
+
+	private const DEFAULT_SYNC_MODE = [
+		'products' => [
+			'pull' => false,
+			'push' => false,
+		],
+		'coupons'  => [
+			'pull' => false,
+			'push' => false,
+		],
+		'shipping' => [
+			'pull' => false,
+			'push' => false,
+		],
+		'settings' => [
+			'pull' => false,
+			'push' => false,
+		],
+	];
+
+	/**
+	 * The base for routes in this controller.
+	 *
+	 * @var string
+	 */
+	protected $route_base = 'sync';
+
+	/**
+	 * SyncController constructor.
+	 *
+	 * @param RESTServer $server
+	 */
+	public function __construct( RESTServer $server ) {
+		parent::__construct( $server );
+	}
+
+	/**
+	 * Registers the routes.
+	 */
+	public function register_routes() {
+		$this->register_route(
+			$this->route_base,
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_sync_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Get the callback function for getting the current sync mode for API PULL.
+	 *
+	 * @return callable
+	 */
+	protected function get_sync_callback(): callable {
+		return function ( Request $request ) {
+			try {
+				$sync_mode = $this->options->get( OptionsInterface::API_PULL_SYNC_MODE );
+				if ( ! $sync_mode ) {
+					$sync_mode = self::DEFAULT_SYNC_MODE;
```

Comment:
Should we be using a check here to ensure the option is an array?

Also isn't it better to always merge with the defaults, with something like `wp_parse_args`. Just in preparation if we would ever add an entry to the defaults it would at least have default values.

Same would apply when we write the endpoint for updating the values.

```suggestion
				if ( ! is_array( $sync_mode ) ) {
					$sync_mode = [];
				}
				
				$sync_mode = wp_parse_args( $sync_mode, self::DEFAULT_SYNC_MODE );
```